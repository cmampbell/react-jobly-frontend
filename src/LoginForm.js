import React, { useState, useEffect } from 'react';
import { Form, useOutletContext } from 'react-router-dom';

const LoginForm = () => {
    const INITIAL_STATE = { username: '', password: '' };
    const [formData, setFormData] = useState(INITIAL_STATE);
    const { login, error, setError } = useOutletContext()

    const handleChange = (evt) => {
        const { name, value } = evt.target
        setFormData(() => ({
            ...formData, [name]: value
        }
        ))
    }

    const handleSubmit = (evt) => {
        evt.preventDefault()
        login(formData)
    }

    // if user navigates away from login screen, clear error
    useEffect(() => () => setError(() => null), [setError])

    return (
        <Form onSubmit={handleSubmit}>
            <label htmlFor='username'>Username:</label>
            <input
                placeholder='username'
                name="username"
                id="username"
                type="text"
                value={formData.username}
                onChange={handleChange} />
            <label htmlFor='password'>Password:</label>
            <input
                placeholder='password'
                name="password"
                id="password"
                type="password"
                value={formData.password}
                onChange={handleChange} />
            {error && <p>{error}</p>}
            <button type="submit">Login</button>
        </Form>
    )
}

export default LoginForm;