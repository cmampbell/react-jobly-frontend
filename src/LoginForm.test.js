import { render } from '@testing-library/react';
import LoginForm from './LoginForm';
import { RouterProvider, createMemoryRouter} from 'react-router-dom';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useOutletContext: function () {
        return {
            login: () => null,
            error: null,
            setError: ()=> null
        }
    }
})
);

const route = [
    {
      path: "/",
      element: <LoginForm />,
    },
  ];

  const router = createMemoryRouter(route);

describe('Smoke and snapshot CompanyCard test', () => {
    it('should render without crashing', () => {
        render(<RouterProvider router={router}/>);
    });
    it('should match snapshot', async ()=> {
        const { asFragment, findByText } = render(<RouterProvider router={router}/>);
        expect(asFragment()).toMatchSnapshot();
        expect(await findByText('Login')).toBeInTheDocument()
    })
})