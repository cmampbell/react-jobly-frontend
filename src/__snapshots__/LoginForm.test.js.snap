// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyCard test should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="/"
    method="get"
  >
    <label
      for="username"
    >
      Username:
    </label>
    <input
      id="username"
      name="username"
      placeholder="username"
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password:
    </label>
    <input
      id="password"
      name="password"
      placeholder="password"
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
